snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/utils/Logger.java

```diff
@@ -62,7 +93,13 @@ public static void track(String tag, String msg, Object... args) {
      */
     public static void e(String tag, String msg, Object... args) {
         if (level >= 1) {
-            Log.e(getTag(tag), getMessage(msg, args));
+            String source = getTag(tag);
+            String message = getMessage(msg, args);
+            if (delegate != null) {
```

Comment:
Would it be worth creating an internal `LoggerDelegate`, and using this when constructing the `Tracker`? This internal delegate could call the `Log...` methods removing the need for this `if`. Makes the code more polymorphic.